disnake/client.py

```diff
@@ -388,10 +413,22 @@ def __init__(
 
         self._hooks: Dict[str, Callable] = {"before_identify": self._call_before_identify_hook}
 
-        self._enable_debug_events: bool = options.pop("enable_debug_events", False)
-        self._enable_gateway_error_handler: bool = options.pop("enable_gateway_error_handler", True)
-        self._connection: ConnectionState = self._get_state(**options)
-        self._connection.shard_count = self.shard_count
+        self._enable_debug_events: bool = enable_debug_events
+        self._enable_gateway_error_handler: bool = enable_gateway_error_handler
+        self._connection: ConnectionState = self._get_state(
+            max_messages=max_messages,
+            application_id=application_id,
+            heartbeat_timeout=heartbeat_timeout,
+            guild_ready_timeout=guild_ready_timeout,
+            allowed_mentions=allowed_mentions,
+            activity=activity,
+            status=status,
+            intents=intents,
+            chunk_guilds_at_startup=chunk_guilds_at_startup,
+            member_cache_flags=member_cache_flags,
+        )
+        self.shard_id = shard_id
+        self._connection.shard_count = self.shard_count = shard_count
```

Comment:
```suggestion
        self.shard_id: Optional[int] = shard_id
        self.shard_count: Optional[int] = shard_count
        self._connection.shard_count = self.shard_count
```
small nit, these attributes should all have annotations

Reply:
Done in 5f5664afe52c96f3fe228260735f46701685b0b2.

Reply:
Due to a glitch in the matrix, the previous statement is a lie. See 1a8337e0a5acab778dded164ba4a4547bd2763df instead. I am very smart.